### PR TITLE
Fix flaky AuthService test

### PR DESCRIPTION
**Pull request by myazinn**
Fixes problems with AuthServiceTest that caused other tests to fail.

### CODE REVIEW

File: services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala [hunk 1]
```diff
@@ -37,9 +36,6 @@ class AuthServiceTest extends WordSpec with Matchers with MockFactory {
 
   import authConfig._
 
-  private val userRepository: UserRepository = stub[UserRepository]
-  private val authUtils: AuthUtils = stub[AuthUtils]
-  private val authService: AuthService = new AuthService(userRepository, authUtils)
```
Comment (myazinn) [inline services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala:42]: Sharing mock between different tests leads to a lot of pain and bugs. I've moved them to `AuthServiceTestContext` below to ensure that each test has it's own mock.

File: services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala [hunk 2]
```diff
@@ -149,23 +145,24 @@ class AuthServiceTest extends WordSpec with Matchers with MockFactory {
     }
 
     "responseFromUser" should {
-      "return AuthResponse" in {
-        (authUtils.getAuthResponse _ when (*, *, *)).returns(Some(any[AuthResponse]))
-        authService.responseFromUser(dummyUser) shouldBe Some(any[AuthResponse])
```
Comment (myazinn) [inline services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala:154]: `any[AuthResponse]` caused problems with other tests. It makes sense, but I have no idea why this test did not fail by itself.
Mockito is great!

File: services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala [hunk 3]
```diff
@@ -175,18 +172,24 @@ class AuthServiceTest extends WordSpec with Matchers with MockFactory {
...
-      "throw the exception if a user isn't active" in {
+      "throw the exception if a user isn't active" taggedAs Service in new AuthServiceTestContext {
         authService.userIsActive(inactiveUser) shouldBe
           Some(InactiveUserException(AuthService.inactiveUserMessage))
       }
     }
   }
 
-  class RefreshTokenContext(lifetime: Long) {
+  class AuthServiceTestContext {
```
Comment (alesavin) [inline services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala:186]: typically this type of fixture-objects is traits (https://www.scalatest.org/user_guide/sharing_fixtures, Instantiating fixture-context objects), but here I see no problem with class

File: services/src/test/scala/cromwell/pipeline/service/ProjectFileServiceTest.scala [hunk 4]
```diff
@@ -28,7 +27,8 @@ class ProjectFileServiceTest extends AsyncWordSpec with Matchers with MockitoSug
     "validateFile" should {
       "return valid message to valid file" taggedAs Service in {
         val request = ProjectFileContent(correctWdl)
-        when(womTool.validate(correctWdl)).thenReturn(Right(any[WomBundle]))
+        val dummyBundle = WomBundle(None, Map.empty, Map.empty, Set.empty)
```
Comment (alesavin) [inline services/src/test/scala/cromwell/pipeline/service/ProjectFileServiceTest.scala:30]: unrelated change?

**Review by GrigoriiBerezin — APPROVED** [review]
If it's really work like that, it's strange :) Thank you for fixing that

File: services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala (hunk 3 above)
Comment (myazinn) [reply services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala:186]: I prefer traits for test fixtures too. I have used class here only because the tests already have a context based on the class.

File: services/src/test/scala/cromwell/pipeline/service/ProjectFileServiceTest.scala (hunk 4 above)
Comment (myazinn) [reply services/src/test/scala/cromwell/pipeline/service/ProjectFileServiceTest.scala:30]: Nope. This test also has a problem with a misplaced matcher `any[Foo]` and might fail eventually.

**Review by alesavin — APPROVED** [review]
LGTM

**Review by iliaonishchenko — APPROVED** [review]
It looks good!
